controllers/cluster_controller_phases.go

```diff
@@ -79,7 +79,7 @@ func (r *ClusterReconciler) reconcileExternal(ctx context.Context, cluster *clus
 	}
 
 	// Initialize the patch helper.
-	patchHelper, err := patch.NewHelper(obj, r.Client)
+	patchHelper, err := patch.NewHelper(obj.DeepCopy(), r.Client)
```

Comment:
The patch helper should already be doing a deep copy of the object where needed.